cmd/gaia/app/genesis.go

```diff
@@ -114,34 +115,38 @@ func GaiaAppGenState(cdc *codec.Codec, appGenTxs []json.RawMessage) (genesisStat
 			err = errors.New("must provide genesis StdTx with exactly 1 CreateValidator message")
 			return
 		}
-		msg := msgs[0].(stake.MsgCreateValidator)
+		_, ok := msgs[0].(stake.MsgCreateValidator)
+		if !ok {
+			err = fmt.Errorf("Genesis transaction %v does not contain a MsgCreateValidator", i)
+			return
+		}
+	}
 
+	for _, acc := range genesisState.Accounts {
 		// create the genesis account, give'm few steaks and a buncha token with there name
-		genaccs[i] = genesisAccountFromMsgCreateValidator(msg, freeFermionsAcc)
-		stakeData.Pool.LooseTokens = stakeData.Pool.LooseTokens.Add(sdk.NewDecFromInt(freeFermionsAcc)) // increase the supply
+		for _, coin := range acc.Coins {
+			if coin.Denom == "steak" {
+				stakeData.Pool.LooseTokens = stakeData.Pool.LooseTokens.
+					Add(sdk.NewDecFromInt(coin.Amount)) // increase the supply
+			}
+		}
 	}
+	genesisState.StakeData = stakeData
+	genesisState.GenTxs = appGenTxs
+	return genesisState, nil
+}
 
-	// create the final app state
-	genesisState = GenesisState{
-		Accounts:     genaccs,
-		StakeData:    stakeData,
+// DefaultGenesisState generates the default state for gaia.
+func DefaultGenesisState() GenesisState {
+	return GenesisState{
```

Comment:
Can we define a function if it doesn't already exist named `NewGenesisState(...)` and use that here instead of populating the struct directly - makes our code less prone to errors in the future as the structs change

Reply:
Done, thanks